src/App.js

```diff
@@ -1,22 +1,46 @@
 import "./utils.css";
-import { Routes, Route } from "react-router-dom";
-import Home from "./Pages/Home/Home";
-import Products from "./Pages/Products/Products";
-import { Navbar, NavAside,Footer} from "./Components/index";
+import { Routes, Route, Navigate } from "react-router-dom";
+import { Home, Products, Login, Signup } from "./Pages";
+import { Navbar, NavAside, Footer } from "./Components/index";
 import { useNav } from "./CustomHooks/useNav";
-import Mockman from 'mockman-js'
+import PrivateRoute from "./PrivateRoute";
+import Mockman from "mockman-js";
+import Cart from "./Pages/Cart/Cart";
+import Wishlist from "./Pages/Wishlist/Wishlist";
+import { useAuth } from "./Context/AuthContext";
 
 export default function App() {
-  const {navToggle,setNavToggle,asideToggleFunction}=useNav()
+    const { navToggle, setNavToggle, asideToggleFunction } = useNav();
+    const {
+        authState: { isAuth },
+    } = useAuth();
     return (
         <div className="App">
             <NavAside navToggle={navToggle} asideToggleFunction={asideToggleFunction} setNavToggle={setNavToggle} />
             <Navbar asideToggleFunction={asideToggleFunction} />
-                <Routes>
-                    <Route path="/" element={<Home />} />
-                    <Route path="/products" element={<Products />} />
-                    <Route path="/mock" element={<Mockman/>}/>
-                </Routes>
+            <Routes>
+                <Route path="/" element={<Home />} />
+                <Route path="/products" element={<Products />} />
+                <Route element={<PrivateRoute />}>
+                    <Route path="/cart" element={<Cart />} />
+                </Route>
+                <Route element={<PrivateRoute />}>
+                    <Route path="/wishlist" element={<Wishlist />} />
+                </Route>
+                {!isAuth ? (
+                    <>
+                        <Route path="/login" element={<Login />} />
+                        <Route path="/signup" element={<Signup />} />
+                    </>
+                ) : (
+                    <>
+                        <Route path="/login" element={<Navigate to="/" />} />
+                        <Route path="/signup" element={<Navigate to="/" />} />
+                    </>
+                )}
+                <Route path="*" element={<Navigate replace to="/" />} />
+                <Route path="/mock" element={<Mockman />} />
+            </Routes>
```

Comment:
suggestion: you can extract the routes into a component

Reply:
Done!